core/chains/evm/logpoller/orm.go

```diff
@@ -0,0 +1,111 @@
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+type ORM struct {
+	chainID *big.Int
+	q       pg.Q
+}
+
+// NewORM creates an ORM scoped to chainID.
+func NewORM(chainID *big.Int, db *sqlx.DB, lggr logger.Logger, cfg pg.LogConfig) *ORM {
+	namedLogger := lggr.Named("ORM")
+	q := pg.NewQ(db, namedLogger, cfg)
+	return &ORM{
+		chainID: chainID,
+		q:       q,
+	}
+}
+
+// InsertBlock is idempotent to support replays.
+func (o *ORM) InsertBlock(h common.Hash, n int64, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	_, err := q.Exec(`INSERT INTO log_poller_blocks (evm_chain_id, block_hash, block_number, created_at) 
+      VALUES ($1, $2, $3, NOW()) ON CONFLICT DO NOTHING`, utils.NewBig(o.chainID), h[:], n)
+	return err
+}
+
+func (o *ORM) SelectBlockByHash(h common.Hash, qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_hash = $1 AND evm_chain_id = $2`, h, utils.NewBig(o.chainID)); err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+func (o *ORM) SelectBlockByNumber(n int64, qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_number = $1 AND evm_chain_id = $2`, n, utils.NewBig(o.chainID)); err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+func (o *ORM) SelectLatestBlock(qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	q := o.q.WithOpts(qopts...)
+	var b LogPollerBlock
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_number = (SELECT MAX(block_number) FROM log_poller_blocks) AND evm_chain_id = $1`, utils.NewBig(o.chainID)); err != nil {
+		return nil, err
+	}
+	return &b, nil
+}
+
+func (o *ORM) DeleteRangeBlocks(start, end int64, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	_, err := q.Exec(`DELETE FROM log_poller_blocks WHERE block_number >= $1 AND block_number <= $2 AND evm_chain_id = $3`, start, end, utils.NewBig(o.chainID))
+	return err
+}
+
+func (o *ORM) DeleteLogs(start, end int64, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	_, err := q.Exec(`DELETE FROM logs WHERE block_number >= $1 AND block_number <= $2 AND evm_chain_id = $3`, start, end, utils.NewBig(o.chainID))
+	return err
+}
+
+// InsertLogs is idempotent to support replays.
+func (o *ORM) InsertLogs(logs []Log, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	_, err := q.NamedExec(`INSERT INTO logs 
+(evm_chain_id, log_index, block_hash, block_number, address, event_sig, topics, tx_hash, data, created_at) VALUES 
+(:evm_chain_id, :log_index, :block_hash, :block_number, :address, :event_sig, :topics, :tx_hash, :data, NOW()) ON CONFLICT DO NOTHING`, logs)
+	return err
+}
+
+func (o *ORM) SelectLogsByBlockRange(start, end int64) ([]Log, error) {
```

Comment:
I was going to suggest indexing this one more directly, but it looks like it is only used for internal tests. Could it be unexported?
```suggestion
func (o *ORM) selectLogsByBlockRange(start, end int64) ([]Log, error) {
```